### PR TITLE
Add initial implementation of singleton operator

### PR DESCRIPTION
**Pull request by enj**
Also moves the controller logic from a builder pattern to a functional options approach.

Signed-off-by: Monis Khan <mkhan@redhat.com>

Builds on #55

/assign @benjaminapetersen

### CODE REVIEW

File: pkg/boilerplate/operator/option.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/service-serving-cert-signer/pkg/boilerplate/controller"
+)
+
+// key is the singleton key shared by all events
+// the value is irrelevant, but pandas are awesome
+const key = "🐼"
```
Comment (benjaminapetersen) [inline pkg/boilerplate/operator/option.go]: I like this, though I wonder if it ever would show up in a log & confuse someone.

**Comment by benjaminapetersen** [comment]
/lgtm 

nice

**Comment by benjaminapetersen** [comment]
ha thought so

File: pkg/boilerplate/operator/option.go (hunk 1 above)
Comment (enj) [reply pkg/boilerplate/operator/option.go]: Sounds like the greatest log message ever :panda_face:

**Comment by smarterclayton** [comment]
reviewed while others are out

**Pull request merged**